fetch_and_process_api_call/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/blocky/as-demo/as"
+)
+
+type Args struct {
+	Market string `json:"market"`
+	CoinID string `json:"coin_id"`
+}
+
+type SecretArgs struct {
+	CoinGeckoAPIKey string `json:"api_key"`
+}
+
+type Price struct {
+	Market    string    `json:"market"`
+	CoinID    string    `json:"coin_id"`
+	Currency  string    `json:"currency"`
+	Price     float64   `json:"price"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+type Output struct {
+	IsErr bool
+	Value Price
+	Error string
+}
+
+func writeOutputToSharedMem(price Price, respErr error) uint64 {
+	isErr := false
+	errString := ""
+	if respErr != nil {
+		isErr = true
+		errString = "Error executing function: " + respErr.Error()
+	}
+
+	output := Output{Value: price, Error: errString, IsErr: isErr}
+	outputData, err := json.Marshal(output)
```

Comment:
use `as.Marshal` here if you're going to keep marshal/unmarshal in `as`

Reply:
done